lib/shared/src/conversion.rs

```diff
@@ -78,68 +60,86 @@ pub fn parse_check_conversion_map(
         }
     }
 
-    parse_conversion_map(types)
+    parse_conversion_map(types, tz)
 }
 
 /// Helper function to parse a mapping of conversion descriptions into actual Conversion values.
 pub fn parse_conversion_map(
     types: &HashMap<String, String>,
+    tz: TimeZone,
 ) -> Result<HashMap<String, Conversion>, ConversionError> {
     types
         .iter()
-        .map(|(field, typename)| {
-            typename
-                .parse::<Conversion>()
-                .map(|conv| (field.clone(), conv))
-        })
+        .map(|(field, typename)| Conversion::parse(typename, tz).map(|conv| (field.clone(), conv)))
         .collect()
 }
 
-#[derive(Debug, Eq, PartialEq, Snafu)]
-pub enum Error {
-    #[snafu(display("Invalid boolean value {:?}", s))]
-    BoolParseError { s: String },
-    #[snafu(display("Invalid integer {:?}: {}", s, source))]
-    IntParseError { s: String, source: ParseIntError },
-    #[snafu(display("Invalid floating point number {:?}: {}", s, source))]
-    FloatParseError { s: String, source: ParseFloatError },
-    #[snafu(display("Invalid timestamp {:?}: {}", s, source))]
-    TimestampParseError { s: String, source: ChronoParseError },
-    #[snafu(display("No matching timestamp format found for {:?}", s))]
-    AutoTimestampParseError { s: String },
-}
-
 impl Conversion {
+    /// Convert the string into a type conversion. The following
+    /// conversion names are supported:
+    ///
+    ///  * `"asis"`, `"bytes"`, or `"string"` => As-is (no conversion)
+    ///  * `"int"` or `"integer"` => Signed integer
+    ///  * `"float"` => Floating point number
+    ///  * `"bool"` or `"boolean"` => Boolean
+    ///  * `"timestamp"` => Timestamp, guessed using a set of formats
+    ///  * `"timestamp|FORMAT"` => Timestamp using the given format
+    ///
+    /// Timestamp parsing does not yet support time zones.
```

Comment:
Is this line of the comment still true?

Reply:
Just drawing attention to this comment again. It seems like this line is no longer valid?

Reply:
Oops, missed that, I will reword.